e2e/types.go

```diff
@@ -148,6 +148,10 @@ type LiveVMValidator struct {
 	// IsShellBuiltIn is a boolean flag which indicates whether or not the command is a shell built-in
 	// that will fail when executed with sudo - requires separate command to avoid command not found error on node
 	IsShellBuiltIn bool
+
+	// IsPodNetwork is a boolean flags which indicates whether or not the validator should run on a pod that is NOT using
+	// host's network interface. For example when testing connectivity from user pods to certain endpoints, we will set it to true
+	IsPodNetwork bool
```

Comment:
Maybe not worth touching now. But I wonder if it's good abstraction. Maybe `LiveVMValidator` can be a function type.

Or `RunScenario` may return a kube clients and we can execute some additional checks in a test function.

Reply:
Yeah I agree we should probably extract this out and either add another type of "PodNetwork" validator or introduce some sort of extra abstraction over the validator concept in e2e. Happy to look into that a bit later!